src/test/java/com/github/martynagil/storemanagement/ProductServiceTest.java

```diff
@@ -0,0 +1,126 @@
+        when(productRepository.findAll()).thenReturn(products);
+
+        List<Product> allProducts = productService.getAllProducts();
+
+        assertThat(allProducts)
+                .hasSize(2)
+                .containsExactly(product1, product2);
+    }
+
+    @Test
+    void shouldRemoveProduct() {
+        Product product = new Product("id","name", "brand", "type", "barcode", 2.09);
+
+        productService.remove(product);
+
+        verify(productRepository).removeById("id");
+    }
+
+    @Test
+    void shouldAddProductToRepository() {
+        Product product = new Product("id","name", "brand", "type", "barcode", 2.09);
+
+        productService.addToRepository(product);
+
+        verify(productRepository).save(product);
+    }
+
+    @Test
+    void shouldSearchByTextInName() {
+        Product product1 = new Product("scsdcs","krem nawilżający", "brand", "type", "barcode", 2.09);
+        Product product2 = new Product("ksdccsc","krem przeciwzmarszczkowy", "brand", "type", "barcode", 2.09);
+        Product product3 = new Product("tonic","tonik", "brand", "type", "barcode", 2.09);
+        List<Product> products = asList(product1, product2, product3);
+        when(productRepository.findAll()).thenReturn(products);
+
+        List<Product> searchedProducts = productService.searchByText("krem");
+
+        assertThat(searchedProducts)
+                .hasSize(2)
+                .containsExactly(product1, product2);
+
+    }
+
+    @Test
+    void shouldSearchByTextInBrand() {
+        Product product1 = new Product("scsdcs","nawilżający", "my super brand", "type", "barcode", 2.09);
+        Product product2 = new Product("ksdccsc","przeciwzmarszczkowy", "my brand is extra", "type", "barcode", 2.09);
+        Product product3 = new Product("tonic","tonik", "company", "type", "barcode", 2.09);
+        List<Product> products = asList(product1, product2, product3);
+        when(productRepository.findAll()).thenReturn(products);
+
+        List<Product> searchedProducts = productService.searchByText("brand");
+
+        assertThat(searchedProducts)
+                .hasSize(2)
+                .containsExactly(product1, product2);
+
+    }
+
+    @Test
+    void shouldSearchByTextInType() {
+        Product product1 = new Product("scsdcs","nawilżający", "my super brand", "typekrem", "barcode", 2.09);
+        Product product2 = new Product("ksdccsc","przeciwzmarszczkowy", "my brand is extra", "krem", "barcode", 2.09);
+        Product product3 = new Product("tonic","tonik", "company", "2type", "barcode", 2.09);
+        List<Product> products = asList(product1, product2, product3);
+        when(productRepository.findAll()).thenReturn(products);
+
+        List<Product> searchedProducts = productService.searchByText("type");
+
+        assertThat(searchedProducts)
+                .hasSize(2)
+                .containsExactly(product1, product3);
+
+    }
+
+    @Test
+    void shouldNotSearchByText() {
```

Comment:
to nie jest dobra nazwa testu, nazwa jest zaprzeczeniem trzech poprzednich. To w końcu powinien znajdować po tekście nie czy nie?

Reply:
No ale to po tym danym tekście nie powinien wyszukać nic